source/_posts/2016-10-18-Understanding-code-signing-for-iOS-apps.md

```diff
@@ -0,0 +1,91 @@
+---
+title:  Understanding code signing for iOS apps
+date: 2016-10-18 23:10:57
+tags: ios,code,signing,security,certificates,apns
+authorId: maco
+categories:
+- iOS
+---
+
+If you're an iOS developer, most likely you've had to deal with code signing. And if you're a junior iOS developer, you might've felt a bit overwhelmed by everything going on in the "Certificates, Identifiers & Profiles" section of the developer portal.
+
+<p align=center><img src="https://d1gwekl0pol55k.cloudfront.net/image/baas/translate_values/hbikk_ospmSpNLyW.gif" /></p>
+
+The goal of this article is to help junior iOS developers understand code signing on a higher level. This will not be a step-by-step tutorial on how to code sign your app. Ideally, after reading this article, you will be able to code sign your app without following any tutorials.
+
+I don't plan to go into lower level details, but we will talk a bit about asymmetric cryptography. 
+
+### Asymmetric cryptography
+The minimum you need to know is that asymmetric cryptography uses a **public key** and a **private key**. The users have to keep their private key for themselves, but they can share the public key. And using those public and private keys, a user can prove that he is indeed himself.
+
+A good high-level explanation of asymmetric cryptography can be found [here](https://blog.vrypan.net/2013/08/28/public-key-cryptography-for-non-geeks/). If you want to know implementation details or the math behind this, you can find them online.
+
+
+### App id
```

Comment:
Shouldn't this be more like "App Id" or even "App ID", I think headings should be always at least camel case.